### PR TITLE
Feature/182 errors controller

### LINKED ISSUE #181: 404 not found error page
> As a user if my trip or route does not exist I want help to know what to do next

There may be times where people are looking for a trip or route that we can't find in the database. In this case we want to show a helpful 404 page that helps people get back on track.

## Acceptance criteria

- [x] I should be aware that there has been a problem
- [x] I should be have a link back to the home page to Plan a new trip

## Notes and links

- [The 404 page large screens](https://www.figma.com/file/VEbdqewes6kebDJcy8aVRX/CamCC---Handover?node-id=0%3A1)
- [404 page on small screens](https://www.figma.com/file/VEbdqewes6kebDJcy8aVRX/CamCC---Handover?node-id=722%3A190)

### LINKED ISSUE #182: Errors Controller
We want to gracefully handle errors in our system. Where possible we should handle errors and provide users with helpful feedback. We want to have a fallback in the case that we havn't caught an error.

Errors might happen for example if someone tries to access a route or trip that has been deleted or not there. In these cases we want to show a helpful 404 page. 

Our system can also throw a generic 500 server error, in these cases too we want to show a helpful error page that helps people get back to their task.

## Acceptance criteria

- [x] It should have a `not_found` action
- [x] When called the `not_found` action should render the errors/not_found template
- [x] When called the `not_found` action should be successful
- [x] It should have an `internal_server_error` action
- [x] When called the `internal_server_error` should render the errors/internal_server_error template
- [x] When called the `internal_server_error` should be successful
- [x] We have a `get` route configured to point to `/404` that should be handled by the `errors#not_found` action
- [x] We have a `get` route configured to point to `/500` that should be handled by the `errors#interna_server_error` action

### LINKED ISSUE #264: Add error handlers to our application controller
All our other controllers inherit their behaviour from the `ApplicationController` this means that methods defined on our application controller will also be present on our other controllers.

We can use this mechanism to centralise how we handle errors in our system. 

## Acceptance criteria

- [x] We have a method on our controller called `render_404`
- [x] Our `render_404` method redirects to the path generated for our `ErrorController`, `not_found` action
- [x] We `rescue_from` the `ActiveRecord::RecordNotFound` error by calling the method `:render_404`

## Notes and links

### PR DESCRIPTION
**Pull request by endyranaudo**
- Generatad `error_page` controller and view for error page
- Created a `not_found` and `internal_server_error` methods
- Created a `route` with `get '/404' and `get '/500'`
- Updated the `application_controller.rb `
- Created a method `render_not_found` in `application_controller.rb ` that redirects to `not_found_path`
- Deleted commented code in `application.scss` from other branches
- Created `errorPage` class in `application.scss`
- Updated tests in `spec/features/error_pages_spec.rb`
- Created page for mobile and web
- Added tests

## Closes 

- Closes #181  issue
- Closes #182 
- Closes #264

### CODE REVIEW

**Comment by endyranaudo** [comment]
I have a little styling issue. Please refer to `application.scss` lines 98 and 107

File: app/views/errors/not_found.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div class='errorPage'>
+  <p class="errorPage__errorNumber"><%= I18n.t('pages.not_found.error_number')%></p>
+  <p class="errorPage__message"><%= I18n.t('pages.not_found.message')%></p>
+  <p class="errorPage__detail"><%= I18n.t('pages.not_found.detail')%></p>
+  <button class='errorPage__button action-link focus:bg-secondary focus:underline active:bg-secondary active:underline'>
```
Comment (mathewtrivett) [inline app/views/errors/not_found.html.erb:5]: @endyranaudo we probably don't want to have an actual button here and instead just provide a styled link back to the home page. Could you update this so it just renders a link with the correct styling?

File: app/views/errors/not_found.html.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<div class='errorPage'>
```
Comment (mathewtrivett) [inline app/views/errors/not_found.html.erb:1]: We also probably want to include some sort of h1 for this page for semantic HTML reasons. My suggestion would probably be the `something is wrong` text as the most appropriate heading here. What do you think?

Do you want to update the markup and any styling if needed

File: config/locales/en.yml [hunk 3]
```diff
@@ -54,6 +54,11 @@ en:
         name: 'Name'
         phone_number: 'Phone number'
       tickbox: 'Tick when you pass here'
+    not_found:
```
Comment (mathewtrivett) [inline config/locales/en.yml:57]: Great use of the I18n locales file!

File: app/views/errors/not_found.html.erb (hunk 2 above)
Comment (mathewtrivett) [inline app/views/errors/not_found.html.erb:1]: We also probably want to include some sort of h1 for this page for semantic HTML reasons. My suggestion would probably be the `something is wrong` text as the most appropriate heading here. What do you think?

Do you want to update the markup and any styling if needed

File: config/locales/en.yml (hunk 3 above)
Comment (mathewtrivett) [inline config/locales/en.yml:57]: Great use of the I18n locales file!

File: spec/features/error_pages_spec.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
```
Comment (mathewtrivett) [inline spec/features/error_pages_spec.rb:3]: Great example of a feature test here, well done Endy.

**Review by mathewtrivett — CHANGES_REQUESTED** [review]
@endyranaudo this all looks great. I've just left you a few comments on the markup for the error page. If you could take a look and make those changes.

After that I think we'll be good to go.

File: app/views/errors/not_found.html.erb (hunk 1 above)
Comment (endyranaudo) [reply app/views/errors/not_found.html.erb:5]: Changed from `button` to `div`

**Review by mathewtrivett — APPROVED** [review]
Great work @endyranaudo let's get this merged in.

**Pull request merged**